site/content/posts/multiple-pinnipeds.md

```diff
@@ -0,0 +1,144 @@
+
+With this system in place, we've achieved our goal. A user can deploy several instances of Pinniped, each interacting only with its own distinct set of API objects.
+
+The default behavior of Pinniped remains unchanged, and we made sure to implement the changes such that they cause little to no overhead when no custom API group has been configured.
+
+### Advantages and Disadvantages
+
+With v0.5.0, each instance of Pinniped to be upgraded and operated 100% independently, with no coordination or shared state needed.
+One remaining constraint is that each instance should be deployed into its own namespace.
+This ensures that any other standard Kubernetes objects such as Secrets and ConfigMaps referenced by the configuration do not overlap.
+
+Our middleware solution carries some ongoing costs:
+
+- It took a non-trivial amount of code to implement all the required transformations.
+  We now have the maintenance burden of ensuring this code continues to work in future versions of the Kubernetes API machinery.
+- Other API consumers (including `kubectl` users) need to know which API group to use.
+  This might be as simple as knowing to run `kubectl get jwtauthenticators.authentication.concierge.team1.example.com`
+  instead of simply `kubectl get jwtauthenticators`.
+  There is no builtin upgrade path between these versions, as there would be with a versioned CRD.
+- The extra encoding/decoding steps cause some performance impact when this feature is in use.
+  None of the Pinniped APIs are used in high throughput use cases, so this was not much a problem for us.
+
+
+## Future Work
+
+We're happy to have shipped this for Pinniped v0.5.0, but we have more ideas about how to extend the concept.
+One idea is to extract the renaming middleware we've written for Pinniped into a standalone Go library that other Kubernetes apps can adopt.
+
+We could also take this a step further and extract the behavior of our middleware into an out-of-process API proxy that can apply these transformations to an unmodified Kubernetes app.
+This would require major changes and it would be challenging to support some features seamlessly, such as Protobuf encoding.
+
+As a team, we have no immediate plans for either of these ideas, but if you are interested please [reach out in GitHub][discussion].
+
+## Join the Pinniped Community!
+Pinniped is better because of our contributors and maintainers.
+It is because of you that we can bring great software to the community.
+Please join us during our online community meetings, occurring every first and third Thursday of the month at 9AM PT / 12PM PT.
+Use [this Zoom link][zoom] to attend and add any agenda items you wish to discuss to [the notes document][meeting-notes].
+Join our [Google Group][google-group] to receive invites to this meeting.
```

Comment:
This section needs some more whitespace, right now it's displayed as a chunk of text. We should also point to the /community page for more info.